lib/download.gi

```diff
@@ -0,0 +1,202 @@
+      FileString( opt.target, res.result );
+      Unbind( res.result );
+    fi;
+    return res;
+  end ) );
+
+Add( Download_Methods, rec(
+  name:= "via SingleHTTPRequest (from the IO package)",
+  isAvailable:= {} -> IsBoundGlobal( "SingleHTTPRequest" ),
+  download:= function( url, opt )
+    local rurl, pos, domain, uri, res;
+
+    if not StartsWith( url, "http://" ) then
+      return rec( success:= false, error:= "protocol is not http" );
+    fi;
+
+    rurl:= ReplacedString( url, "http://", "" );
+    pos:= Position( rurl, '/' );
+    domain:= rurl{ [ 1 .. pos-1 ] };
+    uri:= rurl{ [ pos .. Length( rurl ) ] };
+    if IsBound( opt.target ) and IsString( opt.target ) then
+      res:= ValueGlobal( "SingleHTTPRequest" )( domain, 80, "GET", uri,
+                rec(), false, opt.target );
+    else
+      res:= ValueGlobal( "SingleHTTPRequest" )( domain, 80, "GET", uri,
+                rec(), false, false );
+    fi;
+    if res.statuscode = 0 then
+      return rec( success:= false,
+                  error:= res.status );
+    elif res.statuscode >= 400 then
+      return rec( success:= false,
+                  error:= Concatenation( "HTTP error code ", res.statuscode ) );
+    elif not ( IsBound( opt.target ) and IsString( opt.target ) ) then
+      return rec( success:= true, result:= res.body );
+    else
+      return rec( success:= true );
+    fi;
+  end ) );
+
+Add( Download_Methods, rec(
+  name:= "via wget",
+  isAvailable:= function()
+    local exec;
+
+    exec:= Filename( DirectoriesSystemPrograms(), "wget" );
+    return exec <> fail and IsExecutableFile( exec );
+  end,
+  download:= function( url, opt )
+    local res, outstream, exec, args, code;
+
+    res:= "";
+    outstream:= OutputTextString( res, true );
+    exec:= Filename( DirectoriesSystemPrograms(), "wget" );
+    if IsBound( opt.target ) and IsString( opt.target ) then
+      args:= [ "--quiet", "-O", opt.target, url ];
+    else
+      args:= [ "--quiet", "-O", "-", url ];
+    fi;
+    code:= Process( DirectoryCurrent(), exec, InputTextNone(), outstream, args );
+    CloseStream( outstream );
+    if code <> 0 then
+      return rec( success:= false,
+                  error:= Concatenation( "Process returned ", String( code ) ) );
+    elif not ( IsBound( opt.target ) and IsString( opt.target ) ) then
+      return rec( success:= true, result:= res );
+    else
+      return rec( success:= true );
+    fi;
+  end ) );
+
+Add( Download_Methods, rec(
+  name:= "via curl",
+  isAvailable:= function()
+    local exec;
+
+    exec:= Filename( DirectoriesSystemPrograms(), "curl" );
+    return exec <> fail and IsExecutableFile( exec );
+  end,
+  download:= function( url, opt )
+    local res, outstream, exec, args, code;
+
+    res:= "";
+    outstream:= OutputTextString( res, true );
+    exec:= Filename( DirectoriesSystemPrograms(), "curl" );
+    args:= [ "--silent", "-L", "--fail" ];
+    if IsBound( opt.verifyCert ) and opt.verifyCert = false then
```

Comment:
Again, perhaps better to set the default values uniformly in `Download` ?

Reply:
I've now suggested a change to do that. Then one could apply this patch here:
```suggestion
    if opt.verifyCert = false then
```
or
```suggestion
    if not opt.verifyCert then
```